### PR TITLE
Extend detection validation to solution folder

### PR DESCRIPTION
**Pull request by v-maudan**
Fixes #
- Extend detection validation to solution folder
- Cisco CiscoUmbrella detection present in both folder. (detection and solution folder). Updated logic to keep only solution detection folder.[ If we keep both getting duplicate ID issue (**NoSameTemplateIdTwice**)]

-Updated possible Fix for  CrowdStrike Falcon Endpoint Protection detections.
- Failing due to invalid entity mapping for FileHash entityType

- Updated version parameter to all analytics rules present  in solution folder
## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: .script/tests/detectionTemplateSchemaValidation/DetectionTemplateSchemaValidationTests.cs [hunk 1]
```diff
@@ -107,18 +108,21 @@ public void Validate_DetectionTemplates_TemplatesThatAreInTheWhiteListShouldNotP
...
-            var AllFiles = Directory.GetFiles(detectionPath,"*", SearchOption.AllDirectories).ToList();
-            var numberOfNotYamlFiles = 1; //This is the readme.md file in the directory
-            Assert.True(AllFiles.Count == yamlFiles.Count + numberOfNotYamlFiles,  "All the files in detections folder are supposed to end with .yaml");
+            List<string> detectionPath = DetectionsYamlFilesTestData.GetDetectionPath();
+            var yamlFiles = Directory.GetFiles(detectionPath[0], "*.yaml", SearchOption.AllDirectories).ToList(); // Detection folder
+            yamlFiles.AddRange(Directory.GetFiles(detectionPath[1], "*.yaml", SearchOption.AllDirectories).ToList().Where(s=>s.Contains("Analytic Rules"))); // Extending detection validation to solution folder
+            var AllFiles = Directory.GetFiles(detectionPath[0],"*", SearchOption.AllDirectories).ToList();
+            AllFiles.AddRange(Directory.GetFiles(detectionPath[1], "*", SearchOption.AllDirectories).ToList().Where(s => s.Contains("Analytic Rules")));
+            var numberOfNotYamlFiles = 2; //This is the readme.md file in the detection and Cisco Ise soluton directory.
```
Comment (v-jayakal) [inline .script/tests/detectionTemplateSchemaValidation/DetectionTemplateSchemaValidationTests.cs:116]: @v-maudan, Please remove empty readme file from CIsco ISE of solutions analytics folder and make it 1 (as earlier).

**Review by v-jayakal — CHANGES_REQUESTED** [review]

File: .script/tests/detectionTemplateSchemaValidation/DetectionTemplateSchemaValidationTests.cs (hunk 1 above)
Comment (v-maudan) [reply .script/tests/detectionTemplateSchemaValidation/DetectionTemplateSchemaValidationTests.cs]: done.

File: .script/tests/detectionTemplateSchemaValidation/DetectionsYamlFilesTestData.cs [hunk 2]
```diff
@@ -11,20 +11,36 @@ public class DetectionsYamlFilesTestData : TheoryData<string>
...
+
+        public static string GetRootPath()
+        {
+            var rootDir = Directory.CreateDirectory(GetAssemblyDirectory());
+            var testFolderDepth = 6;
+            for (int i = 0; i < testFolderDepth; i++)
+            {
+                rootDir = rootDir.Parent;
+            }
+            var detectionPath = Path.Combine(rootDir.FullName);
```
Comment (Amitbergman) [inline .script/tests/detectionTemplateSchemaValidation/DetectionsYamlFilesTestData.cs:43]: did you mean to call this variable "rootDirPath"?

File: .script/tests/detectionTemplateSchemaValidation/ValidConnectorIds.json [hunk 3]
```diff
@@ -37,7 +37,9 @@
   "Citrix",
   "CitrixWAF",
   "CloudflareDataConnector",
+  "ContrastProtect",
```
Comment (Amitbergman) [inline .script/tests/detectionTemplateSchemaValidation/ValidConnectorIds.json:40]: which connector is it?

File: .script/tests/detectionTemplateSchemaValidation/ValidConnectorIds.json [hunk 4]
```diff
@@ -58,6 +60,7 @@
   "illusiveAttackManagementSystem",
   "ImpervaWAFGateway",
   "ImportedConnector",
+  "InfobloxCloudDataConnector",
```
Comment (Amitbergman) [inline .script/tests/detectionTemplateSchemaValidation/ValidConnectorIds.json:63]: Where did you get these connector ids from?

File: .script/tests/detectionTemplateSchemaValidation/DetectionsYamlFilesTestData.cs [hunk 5]
```diff
@@ -11,20 +11,36 @@ public class DetectionsYamlFilesTestData : TheoryData<string>
     {
         public DetectionsYamlFilesTestData()
         {
-            string detectionPath = GetDetectionPath();
-            var files = Directory.GetFiles(detectionPath, "*.yaml", SearchOption.AllDirectories).ToList();
+            List<string> detectionPath = GetDetectionPath();
+            var files = Directory.GetFiles(detectionPath[0], "*.yaml", SearchOption.AllDirectories).ToList();
+            files.AddRange(Directory.GetFiles(detectionPath[1], "*.yaml", SearchOption.AllDirectories).ToList().Where(s => s.Contains("Analytic Rules")));
```
Comment (Amitbergman) [inline .script/tests/detectionTemplateSchemaValidation/DetectionsYamlFilesTestData.cs:16]: this filtering should be done in GetDetectionPaths method.

**Review by Amitbergman — CHANGES_REQUESTED** [review]

File: .script/tests/detectionTemplateSchemaValidation/ValidConnectorIds.json (hunk 3 above)
Comment (v-maudan) [reply .script/tests/detectionTemplateSchemaValidation/ValidConnectorIds.json:40]: I have updated these data connector id's present in [solution folder.](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions)

File: .script/tests/detectionTemplateSchemaValidation/ValidConnectorIds.json (hunk 4 above)
Comment (v-maudan) [reply .script/tests/detectionTemplateSchemaValidation/ValidConnectorIds.json:63]: I have updated these data connector id's present in solution folder.

File: .script/tests/detectionTemplateSchemaValidation/DetectionsYamlFilesTestData.cs (hunk 2 above)
Comment (v-maudan) [reply .script/tests/detectionTemplateSchemaValidation/DetectionsYamlFilesTestData.cs]: updated

File: .script/tests/detectionTemplateSchemaValidation/DetectionsYamlFilesTestData.cs (hunk 5 above)
Comment (v-maudan) [reply .script/tests/detectionTemplateSchemaValidation/DetectionsYamlFilesTestData.cs]: Some .yaml file names present in hunting queries as well with same name, hence applying filter in solution folder with "Analytic Rules".

**Comment by v-maudan** [comment]
@Amitbergman , Can you please review this PR ? Thanks.

**Comment by v-maudan** [comment]
@Amitbergman , We need to merge this PR , Can you please review and let us know if any changes are required.

**Review by Amitbergman — APPROVED** [review]

**Pull request merged**